drivers/docker/config.go

```diff
@@ -276,7 +276,7 @@ var (
 			"volume_options": hclspec.NewBlock("volume_options", false, hclspec.NewObject(map[string]*hclspec.Spec{
 				"no_copy": hclspec.NewAttr("no_copy", "bool", false),
 				"labels":  hclspec.NewBlockAttrs("labels", "string", false),
-				"driver_config": hclspec.NewBlockSet("driver_config", hclspec.NewObject(map[string]*hclspec.Spec{
+				"driver_config": hclspec.NewBlock("driver_config", false, hclspec.NewObject(map[string]*hclspec.Spec{
 					"name":    hclspec.NewAttr("name", "string", false),
 					"options": hclspec.NewBlockAttrs("options", "string", false),
 				})),
```

Comment:
Can we create a parsing unit test that fails before and succeeds after this change?

Reply:
Sadly, we don't have a stable pattern for unit testing task driver config; and most config parsing tests are e2e now AFAIK.

I'll resurrect a branch for unit testing https://github.com/hashicorp/nomad/compare/r-testing-hcl-parsing in a separate PR.